dvc/project.py

```diff
@@ -1023,6 +1023,9 @@ def metrics_modify(self, path=None, typ=None, xpath=None):
         self._metrics_modify(path, typ, xpath)
 
     def metrics_add(self, path, typ=None, xpath=None):
+        if not os.path.exists(path):
```

Comment:
What `metrics add` does, is it finds corresponding dvc file that this file is define in, and marks it as `metrics: True`. The problem with https://github.com/iterative/dvc/issues/1316 is that it tries to verify that it is not a binary file in https://github.com/iterative/dvc/blob/0.20.5/dvc/output/local.py#L79 and fails because file doesn't exist. We should not require file to exist and skip that verification, so `dvc metrics add` could work with stages created by `dvc run --no-exec`.